### ISSUE TITLE
Functional gaps for screen reader users - an accessibility issue?

### ISSUE THREAD

**Issue by detlevhfischer** [w3c/matf#81]
In some apps, dialogs (often in the form of bottom sheets) are presented without an explict control to close them. For sighted touch users, tapping outside the dialog will close it. Screen reader users however may be trapped inside the dialog and may only be able to leave it by activating one of the controls inside it. (For keyboard users, ESC will often close the dialog - but what about keyboards without ESC key?). 
In Android the generic back gresture a mirrored "L" will provide the equivalent of ESC: However, I suspect that some screen reader users may not know this gesture.
There are two questions following from this sitruation:

1. Is there any WCAG SC where this issue seems to be covered?  (In the context of EN 301 549, this may belong under 11.5.2.11 List of available actions.) If the answer is none, we should check whether a fitting outcome for WCAG 3 has been drafted.
2. Knowing that screen reader skills / compenence levels will vary a lot across users, what assumptions seem valid when evaluating conformance? Would it be adequate to assume that a SR user will know the generic gesture for "back" so authors can rely on that (omitting a dedicated control, evn if this would be best practice)?

**Comment by TanyaVanWorkum** [comment]
In our testing approach, we assume that users are familiar with common gestures, such as the mirrored L gesture on Android. We would not fail the app if there is a way to close the dialog, even if the user is forced to make a choice (for example, by selecting the same option that is already selected, which triggers closing the dialog).

However, in some rare cases, a user may become truly trapped within the dialog while using a screen reader. At the moment, in the absence of a better alternative, we would fail the app based on SC 2.1.2 (by treating the screen reader similarly to an external keyboard).

Additionally, some interactive elements, such as buttons, receive focus, are included in the focus sequence, but are not operable with the screen reader (i.e., double-tapping does not activate the element). Once again, in the absence of a better alternative, we would fail the app based on SC 2.1.1.

**Comment by detlevhfischer** [comment]
@TanyaVanWorkum wrote:

> Additionally, some interactive elements, such as buttons, receive focus, are included in the focus sequence, but are not operable with the screen reader (i.e., double-tapping does not activate the element). Once again, in the absence of a better alternative, we would fail the app based on SC 2.1.1.

The EN 301 549 has a place for this issue: 11.5.2.12 Execution of available actions

**Comment by jeremy-w** [comment]
System sheets on iOS treat the area outside the sheet as one big control you can focus then activate to dismiss the sheet. The accessibility escape gesture (roughly two-finger Z) also serves similarly to Android’s Back here, and also works to go back in system navigation stacks.

As to user awareness, that gesture is currently (iOS 18.4.1) taught in the second VoiceOver tutorial, “Navigate your iPhone with VoiceOver”. I don’t have an older device handy, but the gesture has been there since at least iOS 5.0 per the [API’s documented availability](https://developer.apple.com/documentation/objectivec/nsobject-swift.class/accessibilityperformescape()/).
